presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkTaskExecutorFactory.java

```diff
@@ -673,6 +667,75 @@ private static OptionalLong computeAllSplitsSize(List<TaskSource> taskSources)
         return OptionalLong.of(sum);
     }
 
+    private void fillNativeExecutionTaskInputs(
+            PlanFragment fragment,
+            PrestoSparkNativeTaskInputs inputs,
+            ImmutableMap.Builder<PlanNodeId, PrestoSparkShuffleReadInfo> shuffleReadInfos)
+    {
+        for (RemoteSourceNode remoteSource : fragment.getRemoteSourceNodes()) {
+            for (PlanFragmentId sourceFragmentId : remoteSource.getSourceFragmentIds()) {
+                PrestoSparkShuffleReadDescriptor shuffleReadDescriptor = inputs.getShuffleReadDescriptors().get(sourceFragmentId.toString());
+
+                if (shuffleReadDescriptor != null) {
+                    shuffleReadInfos.put(remoteSource.getId(), shuffleInfoTranslator.createShuffleReadInfo(shuffleReadDescriptor));
+                }
+            }
+        }
+    }
+
+    private void fillJavaExecutionTaskInputs(
+            PlanFragment fragment,
+            PrestoSparkJavaExecutionTaskInputs inputs,
+            ImmutableMap.Builder<PlanNodeId, List<PrestoSparkShuffleInput>> shuffleInputs,
+            ImmutableMap.Builder<PlanNodeId, List<java.util.Iterator<PrestoSparkSerializedPage>>> pageInputs,
+            ImmutableMap.Builder<PlanNodeId, List<?>> broadcastInputs)
+    {
+        for (RemoteSourceNode remoteSource : fragment.getRemoteSourceNodes()) {
+            List<PrestoSparkShuffleInput> remoteSourceRowInputs = new ArrayList<>();
+            List<java.util.Iterator<PrestoSparkSerializedPage>> remoteSourcePageInputs = new ArrayList<>();
+            List<List<?>> broadcastInputsList = new ArrayList<>();
+            for (PlanFragmentId sourceFragmentId : remoteSource.getSourceFragmentIds()) {
+                Iterator<Tuple2<MutablePartitionId, PrestoSparkMutableRow>> shuffleInput = inputs.getShuffleInputs().get(sourceFragmentId.toString());
+                Broadcast<?> broadcastInput = inputs.getBroadcastInputs().get(sourceFragmentId.toString());
+                List<PrestoSparkSerializedPage> inMemoryInput = inputs.getInMemoryInputs().get(sourceFragmentId.toString());
+
+                if (shuffleInput != null) {
+                    checkArgument(broadcastInput == null, "single remote source is not expected to accept different kind of inputs");
+                    checkArgument(inMemoryInput == null, "single remote source is not expected to accept different kind of inputs");
```

Comment:
The message is same and not descriptive. The entire class has many states and dynamic casts and restraints. 

Can this be refactored to be more maintainable ?

Reply:
As described above.